internal/checks/checks.go

```diff
@@ -465,6 +470,14 @@ func (c *Updater) loop(ctx context.Context) (bool, error) {
 	return connected, errorHandler(err, "getting changes from synthetic-monitoring-api", signalFired)
 }
 
+func (c *Updater) validateProbeCapabilities(capabilities *sm.Probe_Capabilities) error {
+	if !capabilities.DisableScriptedChecks && c.k6Runner == nil {
+		return errCapabilityK6Missing
```

Comment:
is this really causing the agent to exit?

I cannot see why. This is an error, that's true. That should cause the Run function to land in the default case of the switch, on line 309, is that right? If so, isn't it sleeping trying to back off?

Reply:
Good catch, added a case for the new error so we avoid backing off pointlessly

Reply:
I was trying to suggest some changes, but I was having hard time writing it down. I pushed a new commit (511b371be364e9687331808dda9bdcb4f71ab9b4). Could you please review that?